lib/pbench/test/unit/server/test_visualize.py

```diff
@@ -0,0 +1,116 @@
+
+    def test_get_no_dataset(self, query_get_as):
+        response = query_get_as("nonexistent-dataset", "drb", HTTPStatus.NOT_FOUND)
+        assert response.json == {"message": "Dataset 'nonexistent-dataset' not found"}
+
+    def test_dataset_not_present(self, query_get_as):
+        response = query_get_as("fio_2", "drb", HTTPStatus.NOT_FOUND)
+        assert response.json == {
+            "message": "The dataset tarball named 'random_md5_string4' not found"
+        }
+
+    def test_unauthorized_access(self, query_get_as):
+        response = query_get_as("test", "drb", HTTPStatus.FORBIDDEN)
+        assert response.json == {
+            "message": "User drb is not authorized to READ a resource owned by test with private access"
+        }
+
+    def test_successful_get(self, query_get_as, monkeypatch):
+        def mock_extract_data(self, test_name, dataset_name, input_type, data):
+            return {"status": "success", "json_data": "quisby_data"}
+
+        monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
+        monkeypatch.setattr(
+            ApiBase, "_get_dataset_metadata", self.mock_get_dataset_metadata
+        )
+        monkeypatch.setattr(QuisbyProcessing, "extract_data", mock_extract_data)
+
+        response = query_get_as("uperf_1", "test", HTTPStatus.OK)
+        assert response.json["status"] == "success"
+        assert response.json["json_data"] == "quisby_data"
+
+    def test_unsuccessful_get_with_incorrect_data(self, query_get_as, monkeypatch):
+        def mock_find_dataset_with_incorrect_data(self, dataset):
+            class Tarball(object):
+                tarball_path = Path("/dataset/tarball.tar.xz")
+
+                def extract(tarball_path, path):
+                    return "IncorrectData"
+
+            return Tarball
+
+        def mock_extract_data(self, test_name, dataset_name, input_type, data):
+            return {"status": "failed", "exception": "Unsupported Media Type"}
+
+        monkeypatch.setattr(
+            CacheManager, "find_dataset", mock_find_dataset_with_incorrect_data
+        )
+        monkeypatch.setattr(
+            ApiBase, "_get_dataset_metadata", self.mock_get_dataset_metadata
+        )
+        monkeypatch.setattr(QuisbyProcessing, "extract_data", mock_extract_data)
+        response = query_get_as("uperf_1", "test", HTTPStatus.INTERNAL_SERVER_ERROR)
+        response.json["message"] = "Unexpected failure from Quisby"
+
+    def test_unsupported_benchmark(self, query_get_as, monkeypatch):
+        def mock_get_metadata(self, dataset, key):
+            return {"dataset.metalog.pbench.script": "linpack"}
+
+        monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
+        monkeypatch.setattr(ApiBase, "_get_dataset_metadata", mock_get_metadata)
+        response = query_get_as("uperf_1", "test", HTTPStatus.UNSUPPORTED_MEDIA_TYPE)
+        response.json["message"] = "Unsupported Benchmark"
```

Comment:
Consider adding a mock for `QuisbyProcessing.extract_data()` which allows the test to assert that it is _not_ called.